flash_attn/layers/rotary.py

```diff
@@ -92,14 +90,12 @@ def forward(ctx, qkv, cos, sin):
         rotary_dim *= 2
         assert rotary_dim <= headdim
         assert seqlen <= rotary_seqlen
-        assert cos.shape == (seqlen, rotary_dim // 2)
-        assert sin.shape == (seqlen, rotary_dim // 2)
```

Comment:
Can we keep the shape checking for sin?

Reply:
Here I changed the assertion because length of sin might be greater than `seqlen` in general

Reply:
Oh right, that's a great point.